File: 0509-fibonacci-number/0509-fibonacci-number.java
class Solution {
    public int fib(int n) {
        if(n<=1)
            return n;
        
        int prev1 = 1;
        int prev2 = 0;
        int fibo= 0;
        for(int i=2; i<=n; i++){
            fibo = prev1 + prev2;
            prev2 = prev1;
            prev1 = fibo;
        }
        return fibo;
    }
}

File: 0509-fibonacci-number/0509-fibonacci-number.py
class Solution:
    def fib(self, n: int) -> int:
        ans = [0]*(n+1)
        
        if n < 2:
            return n
        prev =1
        prev2 = 0

        for i in range(2,n+1):
            temp = prev + prev2
            prev2 = prev
            prev = temp
        return prev
        
